docs/C_Sharp.md

```diff
@@ -0,0 +1,46 @@
+# File Metrics for C#
+
+This document outlines the definition of file metrics for C#.
+
+### complexity
+
+The "complexity" metric counts:
+
+-   `if` and `else if` statements
+-   Uses of the ternary operator `condition ? then : otherwise`
+-   Null-coalescing operators `??` and `??=`
+-   Loops, including
+    -   `for` loops and `foreach` loops
+    -   `while` loops
+    -   `do-while` loops
+-   `case` labels in switch-statements
+-   Not-default switch expression arms in switch expressions `x switch { value1 => expression1, ...}`
+-   `catch` blocks
+-   logical binary operations `&&` and `||`
+-   pattern combinators `and` and `or` in `is`-expression, switch statements and switch expressions
+-   everything counted for the ["functions" metric](#functions)
+
+It does **not** count:
+
+-   case guards with the keyword `when`
```

Comment:
why not?

Reply:
lets discuss if we count case guards for complexity in every language in other issue #88